src/comm/MAVLinkProtocol.cc

```diff
@@ -364,7 +364,7 @@ void MAVLinkProtocol::receiveBytes(LinkInterface* link, QByteArray b)
 
             // Anyone handling the message could close the connection, which deletes the link,
             // so we check if it's expired
-            if (1 == linkPtr.use_count()) {
+            if (!linkPtr.isNull()) {
```

Comment:
This does not look correct, it should use the ref count.

Reply:
There is no function to get the reference count for a QSharedPointer
https://doc.qt.io/qt-5/qsharedpointer.html
https://stackoverflow.com/questions/41934716/reference-count-number-for-qsharedpointer

Reply:
Yes I saw, so we can't use it.

Reply:
Are you suggesting we can't use QSharedPointer because it doesn't have a reference count function? This doesn't seem like a very good reason, especially since the places where it's used don't really add any value.

Reply:
I'm not sure what you mean by this. The logic here actually prevents subtle crashes, so it's very relevant and changing the semantics breaks it.

Reply:
oh I overlooked the details. Hmm is there actually an instance in the code base where a slot handling  `messageReceived` actually deletes the link? Doesn't seem quite right. Anyway this PR should be good to go now.

Reply:
Yes I think when rebooting the vehicle if I remember correctly. There's a lot that can happen within receiving messages, and thus it's quite complex.